homeassistant/components/tradfri/__init__.py

```diff
@@ -147,10 +154,21 @@ async def on_hass_stop(event):
             hass.config_entries.async_forward_entry_setup(entry, component)
         )
 
+    async def async_keep_alive(now):
+        try:
+            await api(gateway.get_gateway_info())
+        except RequestError:
+            _LOGGER.error("Keep-alive failed")
+            await factory.shutdown()
```

Comment:
I'm not sure we should shutdown the factory. We can't recover from that unless we create a new factory.

The factory will reset the protocol upon error before raising it to us, so I think we can just log the error and keep going.

Reply:
Cool, thanks!
I think you're right, and I've pushed the change to the code.